src/core/umount.c

```diff
@@ -381,13 +381,19 @@ static bool nonunmountable_path(const char *path) {
                 || path_startswith(path, "/run/initramfs");
 }
 
+/* This includes remounting readonly, which changes the kernel mount options.
+ * Therefore the list passed to this function is invalidated, and should not be reused. */
+
 static int mount_points_list_umount(MountPoint **head, bool *changed, bool log_error) {
         MountPoint *m, *n;
         int n_failed = 0;
 
         assert(head);
 
         LIST_FOREACH_SAFE(mount_point, m, n, *head) {
```

Comment:
if we don't modify the list anymore while iterating through it, let's simplify this, and use LIST_FOREACH() instead of LIST_FOREACH_SAFE(), and we can drop one temporary variable...